pkg/services/ngalert/remote/remote_primary_forked_alertmanager.go

```diff
@@ -37,13 +37,18 @@ func (fam *RemotePrimaryForkedAlertmanager) ApplyConfig(ctx context.Context, con
 	return nil
 }
 
-// TODO: save the new configuration hash in memory.
 func (fam *RemotePrimaryForkedAlertmanager) SaveAndApplyConfig(ctx context.Context, config *apimodels.PostableUserConfig) error {
 	return nil
 }
 
-// TODO: save the new configuration hash in memory.
 func (fam *RemotePrimaryForkedAlertmanager) SaveAndApplyDefaultConfig(ctx context.Context) error {
+	if err := fam.remote.SaveAndApplyDefaultConfig(ctx); err != nil {
+		return fmt.Errorf("failed to send the default configuration to the remote Alertmanager: %w", err)
+	}
+
+	if err := fam.internal.SaveAndApplyDefaultConfig(ctx); err != nil {
+		fam.log.Error("Error applying the default configuration to the internal Alertmanager", "err", err)
+	}
```

Comment:
Just for my understanding - the intention is that if saving to the remote and fails, then don't update the internal, correct?

Reply:
Yep!